### DISCUSSION TITLE
v3.5.0-M5

### DISCUSSION THREAD

**Discussion by nafg** [slick/slick#2862] [Announcements]
### Enhancements ✨
* Restore ability to concat sql strings, closes #1161 by @OlegYch in https://github.com/slick/slick/pull/2796
* MappedProjection: make the source type no longer part of the API by @nafg in https://github.com/slick/slick/pull/2854
* Fix `getOrElse` type inference issue in Scala 3, closes #2761 by @OlegYch in https://github.com/slick/slick/pull/2803
### Bugs fixed 🐛️
* Fixed behavior of `insertOrUpdate` on tables with auto_increment option by @jkugiya in https://github.com/slick/slick/pull/2785
* Replace filter on aggregates with CASE, closes #2235 by @OlegYch in https://github.com/slick/slick/pull/2832
### Documentation improvements 📃️
* Document tupled workaround for Scala 3 by @mdedetrich in https://github.com/slick/slick/pull/2760
* Fix scaladoc -sourcepath issues by @nafg in https://github.com/slick/slick/pull/2769
* schemas.md: improve Mapped Tables section by @nafg in https://github.com/slick/slick/pull/2770
* upgrade.md: Add initial 3.5.0 section by @nafg in https://github.com/slick/slick/pull/2773
* upgrade.md: fix sections by @nafg in https://github.com/slick/slick/pull/2805
* upgrade.md: mention the change to MappedProjection from PR #2854 by @nafg in https://github.com/slick/slick/pull/2860
### Chores 🛠
* fix warning in `TransactionTest` by @xuwei-k in https://github.com/slick/slick/pull/2762
* Update dependency com.lightbend.paradox:sbt-paradox to v0.10.4 by @renovate in https://github.com/slick/slick/pull/2764
* Update dependency com.lightbend.paradox:sbt-paradox to v0.10.5 by @renovate in https://github.com/slick/slick/pull/2765
* Update paradox-theme-generic, sbt-paradox to 0.10.5 by @slick-scala-steward in https://github.com/slick/slick/pull/2766
* Try Scala 3.3.1-RC3 by @nafg in https://github.com/slick/slick/pull/2758
* Update scala3-library to 3.3.1-RC4 by @slick-scala-steward in https://github.com/slick/slick/pull/2768
* build.sbt: default scala version to 2.13 by @nafg in https://github.com/slick/slick/pull/2771
* CompatReportPlugin: Remove `<detail>/<summary>` by @nafg in https://github.com/slick/slick/pull/2772
* build.sbt: buildCompatReport: include header by @nafg in https://github.com/slick/slick/pull/2775
* Update dependency com.mysql:mysql-connector-j to v8.1.0 by @renovate in https://github.com/slick/slick/pull/2777
* Update sbt-version-policy to 2.1.2 by @slick-scala-steward in https://github.com/slick/slick/pull/2778
* Update dependency com.typesafe:sbt-mima-plugin to v1.1.3 by @renovate in https://github.com/slick/slick/pull/2780
* Update sbt-mima-plugin to 1.1.3 by @slick-scala-steward in https://github.com/slick/slick/pull/2781
* Update dependency sbt/sbt to v1.9.3 by @renovate in https://github.com/slick/slick/pull/2786
* Update sbt-version-policy to 2.1.3 by @slick-scala-steward in https://github.com/slick/slick/pull/2787
* Update logback-classic to 1.3.9 by @slick-scala-steward in https://github.com/slick/slick/pull/2789
* Update logback-classic to 1.3.10 by @slick-scala-steward in https://github.com/slick/slick/pull/2792
* Update logback-classic to 1.3.11 by @slick-scala-steward in https://github.com/slick/slick/pull/2794
* Update scala3-library to 3.3.1-RC5 by @slick-scala-steward in https://github.com/slick/slick/pull/2797
* Update ojdbc8_g to 21.11.0.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2798
* Explicitly handle this.type soundness reported as error since Scala 3.4.x by @WojciechMazur in https://github.com/slick/slick/pull/2801
* Only use scala-collection-compat for Scala 2.12 by @mdedetrich in https://github.com/slick/slick/pull/2747
* Fix incorrect POM filter with case sensitivity by @mdedetrich in https://github.com/slick/slick/pull/2804
* Update scala3-library to 3.3.1-RC6 by @slick-scala-steward in https://github.com/slick/slick/pull/2806
* Add scalac inliner options for Scala 2 by @mdedetrich in https://github.com/slick/slick/pull/2808
* Update sbt to 1.9.4 by @slick-scala-steward in https://github.com/slick/slick/pull/2812
* Update scala3-library to 3.3.1-RC7 by @slick-scala-steward in https://github.com/slick/slick/pull/2813
* Update dependency org.slf4j:slf4j-api to v2.0.8 by @renovate in https://github.com/slick/slick/pull/2817
* Update slf4j-api to 2.0.8 by @slick-scala-steward in https://github.com/slick/slick/pull/2816
* Update dependency org.slf4j:slf4j-api to v2.0.9 by @renovate in https://github.com/slick/slick/pull/2819
* Update actions/checkout action to v4 by @renovate in https://github.com/slick/slick/pull/2820
* Update dependency org.scalatestplus:testng-7-5 to v3.2.17.0 by @renovate in https://github.com/slick/slick/pull/2821
* Update scala3-library to 3.3.1 by @slick-scala-steward in https://github.com/slick/slick/pull/2823
* Update dependency org.scoverage:sbt-scoverage to v2.0.9 by @renovate in https://github.com/slick/slick/pull/2824
* add explicit type by @xuwei-k in https://github.com/slick/slick/pull/2825
* Update dependency sbt/sbt to v1.9.5 by @renovate in https://github.com/slick/slick/pull/2829
* Update sbt to 1.9.5 by @slick-scala-steward in https://github.com/slick/slick/pull/2830
* Update dependency sbt/sbt to v1.9.6 by @renovate in https://github.com/slick/slick/pull/2833
* Update sbt to 1.9.6 by @slick-scala-steward in https://github.com/slick/slick/pull/2834
* Hide old compatibility changes report by @xuwei-k in https://github.com/slick/slick/pull/2840
* add explicit type by @xuwei-k in https://github.com/slick/slick/pull/2838
* Revert "Hide old compatibility changes report" by @nafg in https://github.com/slick/slick/pull/2841
* Test on JDK 21 instead of 17 and fix  by @xuwei-k in https://github.com/slick/slick/pull/2839
* Scala 2.13.12 by @xuwei-k in https://github.com/slick/slick/pull/2842
* Update typesafe:config to 1.4.3 by @slick-scala-steward in https://github.com/slick/slick/pull/2844
* Update dependency sbt/sbt to v1.9.7 by @renovate in https://github.com/slick/slick/pull/2846
* Update sbt to 1.9.7 by @slick-scala-steward in https://github.com/slick/slick/pull/2847
* Update dependency com.mysql:mysql-connector-j to v8.2.0 by @renovate in https://github.com/slick/slick/pull/2850
* Update actions/github-script action to v7 by @renovate in https://github.com/slick/slick/pull/2857
* Update dependency org.xerial.sbt:sbt-sonatype to v3.10.0 by @renovate in https://github.com/slick/slick/pull/2859
* Update dependency org.postgresql:postgresql to v42.7.0 by @renovate in https://github.com/slick/slick/pull/2861

## New Contributors
* @WojciechMazur made their first contribution in https://github.com/slick/slick/pull/2801

**Full Changelog**: https://github.com/slick/slick/compare/v3.5.0-M4...v3.5.0-M5

<hr /><em>This discussion was created from the release <a href='https://github.com/slick/slick/releases/tag/v3.5.0-M5'>v3.5.0-M5</a>.</em>

**Comment by fabienfoerster** [comment]
Hi, do we have a date for the stable release ?
